wgengine/ifstatus_windows.go

```diff
@@ -0,0 +1,102 @@
+// Copyright (c) 2021 Tailscale Inc & AUTHORS All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package wgengine
+
+import (
+	"fmt"
+	"github.com/tailscale/wireguard-go/tun"
+	"golang.zx2c4.com/wireguard/windows/tunnel/winipcfg"
+	"sync"
+	"tailscale.com/types/logger"
+	"time"
+)
+
+type ifaceWatcher struct {
+	lck  sync.Mutex
+	luid winipcfg.LUID
+	done bool
+	sig  chan bool
+}
+
+func (iw *ifaceWatcher) callback(notificationType winipcfg.MibNotificationType, iface *winipcfg.MibIPInterfaceRow) {
+	// Probably should check only when MibParameterNotification, but just in case included MibAddInstance also.
+	if notificationType == winipcfg.MibParameterNotification || notificationType == winipcfg.MibAddInstance {
+		go iw.isUp()
+	}
+}
+
+func (iw *ifaceWatcher) isUp() bool {
+	iw.lck.Lock()
+	defer iw.lck.Unlock()
+
+	if iw.done {
+		// We already know that it's up
+		return true
+	}
+
+	s, err := iw.getOperStatus()
+	if err != nil || s != winipcfg.IfOperStatusUp {
```

Comment:
do we care about logging this error?

Reply:
hmmm... will not hurt if we log. Will add.